plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/data/MapboxViewportDataSource.kt

```diff
@@ -0,0 +1,272 @@
+package com.mapbox.maps.plugin.viewport.data
+
+import android.location.Location
+import androidx.annotation.UiThread
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.delegates.MapCameraManagerDelegate
+import com.mapbox.maps.plugin.delegates.MapTransformDelegate
+import com.mapbox.maps.plugin.viewport.data.ViewportDataSourceProcessor.getMapAnchoredPaddingFromUserPadding
+import com.mapbox.maps.plugin.viewport.data.ViewportDataSourceProcessor.getScreenBoxForFraming
+import com.mapbox.maps.plugin.viewport.data.ViewportDataSourceProcessor.getSmootherBearingForMap
+import com.mapbox.maps.plugin.viewport.data.ViewportProperty.BearingProperty
+import com.mapbox.maps.plugin.viewport.data.ViewportProperty.CenterProperty
+import com.mapbox.maps.plugin.viewport.data.ViewportProperty.PitchProperty
+import com.mapbox.maps.plugin.viewport.data.ViewportProperty.ZoomProperty
+import com.mapbox.maps.plugin.viewport.transition.normalizeBearing
+import com.mapbox.maps.toCameraOptions
+import java.util.concurrent.CopyOnWriteArraySet
+import kotlin.math.max
+import kotlin.math.min
+
+/**
+ * Mapbox default implementation for ViewportDataSource.
+ */
+@UiThread
+class MapboxViewportDataSource(
```

Comment:
guess should be internal

Reply:
As mentioned in the PR's description. Currently the activity is accessing this class directly, we need to refactor and abstract the public interface from it in the follow-up PRs.